src/reader/mod.rs

```diff
@@ -593,15 +605,18 @@ impl<R> Reader<R> {
     /// Read text into the given buffer, and return an event that borrows from
     /// either that buffer or from the input itself, based on the type of the
     /// reader.
-    fn read_event_impl<'i, B>(&mut self, buf: B) -> Result<Event<'i>>
+    fn read_event_impl<'i, B>(&mut self, mut buf: B) -> Result<Event<'i>>
     where
         R: XmlSource<'i, B>,
     {
         read_event_impl!(self, buf, self.reader, read_until_open, read_until_close)
     }
 
     /// Read until '<' is found, moves reader to an `OpenedTag` state and returns a `Text` event.
-    fn read_until_open<'i, B>(&mut self, buf: B) -> Result<Event<'i>>
+    ///
+    /// Returns inner Ok if the loop should be broken and an event returned.
+    /// Returns inner Err with the same [buf] because Rust borrowck stumbles upon this case in particular.
```

Comment:
```suggestion
    /// Returns inner `Ok` if the loop should be broken and an event returned.
    /// Returns inner `Err` with the same `buf` because Rust borrowck stumbles upon this case in particular.
```